zone/attack.cpp

```diff
@@ -2860,8 +2861,15 @@ bool NPC::Death(Mob* killer_mob, int64 damage, uint16 spell, EQ::skills::SkillTy
 		}
 
 		entity_list.LimitRemoveNPC(this);
+
 		entity_list.AddCorpse(corpse, GetID());
 
+		// The client sees NPC corpses as name's_corpse.  The server uses
+		// name`s_corpse so that %T works on corpses (client workaround)
+		// Rename the new corpse on client side.
+		std::string old_name = Strings::Replace(corpse->GetName(), "`s_corpse", "'s_corpse");
+		SendRename(killer_mob, old_name.c_str(), corpse->GetCleanName());
```

Comment:
This should set the new name to `corpse->GetName()` instead of the clean name. The new name on the client should still be in the raw ``name`s_corpse##`` format

Reply:
I did exactly what @neckkola did - sending the name without and it works without a hitch.  Are you sure?  @neckkola ?

I saw it was different from what you suggested, but my solution (before we moved it into a function) and @neckkola both use GetCleanName(). 

Let me know.

Reply:
Yes, we want to just replace `Some_Name's_corpse01` with ``Some_Name`s_corpse01`` on the client here. Using clean name would just replace it with ``Some Name`s corpse``. The client will handle trimming and stripping underscores and numbers for any UI/%t output.

There's places in the client that get spawns by name and this helps distinguish them (for example using a non-mq `/target Some_Name's_corpse01` vs `/target Some_Name's_corpse02`). This would also allow scripts to use `Mob::TempName` on the corpse afterwards since it renames using the server side raw name which the client uses to find the right spawn in case there's multiples with the same name.

Reply:
Got it - will fix in the next couple of days.  Tomorrow is crazy and I of course want to retest.

Reply:
Fixed.  Thanks.  Was dumb of me not to see that.  I was so tired of then many iterations of this that I was sloppy. 

This is by far the best solution so far, so thank you.